src/main/java/bio/terra/stairway/Database.java

```diff
@@ -0,0 +1,460 @@
+                        "is incompatible with software schema version " + FLIGHT_SCHEMA_VERSION);
+            }
+
+        } catch (SQLException ex) {
+            throw new DatabaseSetupException("Failed to get schema version state", ex);
+        }
+    }
+
+    // Database accessors - package scoped so unit tests can use them
+    boolean versionTableExists(Statement statement) throws SQLException {
+        String query = "SELECT COUNT(*) AS version_table_count" +
+                " FROM information_schema.tables WHERE table_schema = '" +
+                (schemaName == null ? "public" : schemaName) +
+                "' AND table_name = '" +
+                (nameStem == null ? "" : nameStem + '_') + FLIGHT_VERSION_TABLE + "'";
+
+        try (ResultSet rs = statement.executeQuery(query)) {
+            if (rs.next()) {
+                int tableCount = rs.getInt("version_table_count");
+                return (tableCount == 1);
+            } else {
+                throw new DatabaseSetupException("Invalid result from information_schema query");
+            }
+        }
+    }
+
+    String readDatabaseSchemaVersion(Statement statement) throws SQLException {
+        try (ResultSet rs = statement.executeQuery("SELECT version FROM " + flightVersionTableName)) {
+            if (rs.next()) {
+                return rs.getString("version");
+            }
+            throw new DatabaseSetupException("Schema version not found");
+        }
+    }
+
+    String readDatabaseSchemaCreateTime(Statement statement) throws SQLException {
+        try (ResultSet rs = statement.executeQuery(
+                "SELECT create_time::text AS createtime FROM " + flightVersionTableName)) {
+            if (rs.next()) {
+                return rs.getString("createtime");
+            }
+            throw new DatabaseSetupException("Schema create time not found");
+        }
+    }
+
+    /**
+     * Create the schema and store its version number
+     */
+    private void create() {
+        if (isDatabaseDisabled()) {
+            return;
+        }
+
+        try (Connection connection = dataSource.getConnection();
+             Statement statement = connection.createStatement()) {
+            connection.setAutoCommit(false);
+
+            if (schemaName != null) {
+                statement.executeUpdate("CREATE SCHEMA IF NOT EXISTS " + schemaName);
+            }
+
+            statement.executeUpdate(
+                    "CREATE TABLE " + flightVersionTableName +
+                            "(version VARCHAR(50)," +
+                            " create_time TIMESTAMP)");
+
+            statement.executeUpdate(
+                    "INSERT INTO " + flightVersionTableName +
+                            " VALUES('" + FLIGHT_SCHEMA_VERSION + "', CURRENT_TIMESTAMP)");
+
+            statement.executeUpdate(
+                    "CREATE TABLE " + flightTableName +
```

Comment:
Why not handle schema creation and modifications with liquibase?

Reply:
See my general reply to Andrea's general comment...

Reply:
To be done in follow-on PR